docs/source/development/reference/functions.rst

```diff
@@ -521,6 +521,9 @@ Lists
     entry of the returned array is a NULL-entry. The array is statically
     allocated and does not need to be freed after use.
 
+    Note: starting with PROJ 7.1, this function is deprecated by
```

Comment:
How about also adding `__attribute__ ((deprecated))` to the function definition in `proj.h` ?

Reply:
Added